### PR TITLE
CORDA-4025: Durable streams changes to OS

### PR DESCRIPTION
**Pull request by vkolomeyko**
Please see linked Jira for the context for this change.

This is a pre-requisite for: corda/enterprise#3816

### CODE REVIEW

File: .ci/api-current.txt [hunk 1]
```diff
@@ -155,8 +155,8 @@ public final class net.corda.core.context.AuthServiceId extends java.lang.Object
...
   public <init>(net.corda.core.context.InvocationOrigin, net.corda.core.context.Trace, net.corda.core.context.Actor, net.corda.core.context.Trace, net.corda.core.context.Actor)
   public <init>(net.corda.core.context.InvocationOrigin, net.corda.core.context.Trace, net.corda.core.context.Actor, net.corda.core.context.Trace, net.corda.core.context.Actor, int, kotlin.jvm.internal.DefaultConstructorMarker)
-  public <init>(net.corda.core.context.InvocationOrigin, net.corda.core.context.Trace, net.corda.core.context.Actor, net.corda.core.context.Trace, net.corda.core.context.Actor, java.util.List<?>, String)
-  public <init>(net.corda.core.context.InvocationOrigin, net.corda.core.context.Trace, net.corda.core.context.Actor, net.corda.core.context.Trace, net.corda.core.context.Actor, java.util.List, String, int, kotlin.jvm.internal.DefaultConstructorMarker)
+  public <init>(net.corda.core.context.InvocationOrigin, net.corda.core.context.Trace, net.corda.core.context.Actor, net.corda.core.context.Trace, net.corda.core.context.Actor, java.util.List<?>, String, net.corda.core.internal.context.DurableStreamContext)
```
Comment (dimosr) [inline .ci/api-current.txt:158]: I think that's technically an API breakage we could avoid by creating an extra constructor. 

However, we had to do it in other places and it ended up making the code quite hard to maintain. This type also looks like something that's probably going to be initialised mostly by the platform and used in a read-only way from customers, which means these breakages might not be really impactful in practice. So, if we get a confirmation we can accept this, I am happy with it.

**Review by dimosr — COMMENTED** [review]
Code changes look good, just a comment on the API changes.

File: .ci/api-current.txt (hunk 1 above)
Comment (vkolomeyko) [reply .ci/api-current.txt:158]: > these breakages might not be really impactful in practice

Yes, I agree.
Also, it is rather unfortunate that class `net.corda.core.context.InvocationContext` is included into client facing API. IMO it should be and `internal` class  in `:node` module.

**Review by rick-r3 — COMMENTED** [review]
LGTM but I guess we're not merging.

**Comment by vkolomeyko** [comment]
Closing as these changes been merged to C5 as: https://github.com/corda/corda5/pull/53

**Pull request closed without merging**